azurerm/resource_arm_function_app.go

```diff
@@ -556,6 +577,21 @@ func resourceArmFunctionAppRead(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
+	functions, err := client.ListFunctionsComplete(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+	getSecretsFunc := func(functionName string) (web.FunctionSecrets, error) {
```

Comment:
Instead of doing this, could we just pass in the required variables (client ctx etc) into the flatten function?

Reply:
Good call, got overexcited about being back in a functional language after time in Java. Removed.